### DISCUSSION TITLE
Alpha bug with 7.0.10-53

### DISCUSSION THREAD

**Discussion by GeeMack** [ImageMagick/ImageMagick#3063] [Development] [answered]
ImageMagick 7.0.10-53 Q16 x64 HDRI on Windows 10.

When I run this command...

    magick rose: -bordercolor %[pixel:p{20,20}] -border 12 result.png

I get this result, as I would expect, the rose image with a border of the color of the pixel at location x20 y20.

![result](https://user-images.githubusercontent.com/65475279/103452652-34877f80-4c97-11eb-89cb-1edcdcd253df.png)

I've come across a problem with v7.0.10-53 when I do anything that adds an alpha channel. By appending a transparent pixel to the rose image with this command...

    magick rose: xc:none +append -bordercolor %[pixel:p{20,20}] -border 12 result.png

![result](https://user-images.githubusercontent.com/65475279/103452662-56810200-4c97-11eb-8b52-a50d072a9caa.png)

The result has a transparent border instead of the red border expected.

This transparent effect happens in any case where I use "%[pixel:p{x,y}]" to acquire a color and apply it to any image with an alpha channel. I don't know when this issue first appeared, but it did not occur in version 7.0.10-44 or before.

**Comment by dlemstra** [comment]
It looks like this was broken due to this commit: 3ba48634da6d939725b522ee5b0fcb25a960a53f. The problem can also be reproduced with:
`magick -size 100x100 xc:srgba(169,59,42,1) result.png`. Due to that change it now expects the alpha value to be in the quantum range, where `1` is almost transparent. We will need to investigate this futher.

**Comment by urban-warrior** [comment, marked as answer]
hanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://www.imagemagick.org/download/beta/ by sometime tomorrow.

**Comment by GeeMack** [comment]
The issues associated with this bug seem to be corrected in version ImageMagick 7.0.10-54 Q16 x64. Thanks!
